api/v1alpha8/openstackcluster_types.go

```diff
@@ -167,6 +167,8 @@ type OpenStackClusterSpec struct {
 
 // OpenStackClusterStatus defines the observed state of OpenStackCluster.
 type OpenStackClusterStatus struct {
+	// Ready is true when the cluster infrastructure is ready.
+	// +optional
```

Comment:
I highly suspect that we need something like this so the OpenStackCluster can be written when a reconcile finishes.